src/logger.ts

```diff
@@ -0,0 +1,25 @@
+import { createDetailedLogger } from './detailed.logger';
+import { createInlineLogger } from './inline.logger';
+import { createJSONLogger } from './json.logger';
+import { LogFormat } from './log-format.enum';
+import { LogLevel } from './log-level.enum';
+import { LoggerOptions } from './logger-options.interface';
+
+export const DEFAULT_OPTIONS: LoggerOptions = {
+  format: LogFormat.JSON,
+  level: LogLevel.INFO,
+};
+
+export const createLogger = (options: Partial<LoggerOptions> = {}) => {
+  const { format, level } = { ...DEFAULT_OPTIONS, ...options };
```

Comment:
this is a bug. please use lodash merge (or something similar) to do this.

```ts
createLogger({ format: undefined }); // doesn't use `DEFAULT_OPTIONS` because the `format` property was overridden
```

we should also add a regression test for this specific case